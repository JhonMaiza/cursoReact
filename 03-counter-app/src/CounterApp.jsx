import Prototype from  'prop-types';
import { useState } from 'react';
export const CounterApp = ( { value } ) => {

  const [ counter, setCounter ] = useState(value)

  const handleAdd = () => {
    setCounter( counter + 1 );
    //setCounter( c => c + 1 )
  }
  const handleSubtract = () => {
    setCounter( counter - 1 );
  }

  const handleRestart = () => {
    setCounter( value );
  }
  return (
    <>
      <h1> Counter App </h1>
      <h2>{ counter }</h2>
      <button onClick={handleAdd}>
        Aumentar
      </button>

      <button onClick={handleSubtract}>
        Restar
      </button>

      <button aria-label='btn-reset' onClick={handleRestart}>
        Restablecer
      </button>
    </>
  )
}

CounterApp.propTypes = {
  value: Prototype.number.isRequired,
}

// CounterApp.defaultProps = {
//   value: 12,
// }
