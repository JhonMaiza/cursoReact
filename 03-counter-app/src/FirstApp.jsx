import PropTypes from 'prop-types';

// si no hay dependencia de los componente se debe sacar del funcional component
const funcion = (a, b) => {
  console.log('hola');
  return a + b;
}

const newMessage = {
  message: 'hola',
  titulo: 'Jhon'
}

export const FirstApp = ( { title, subtitle, nombre} ) => {
  
  return (
    <>
      {/* <code>{ JSON.stringify(newMessage)</code>  } */}
      <h1 data-testid="test-title">{ title }</h1>
      <p data-testid="test-subtitle">{subtitle}</p>
      <p data-testid="test-subtitle">{subtitle}</p>
      <p>{nombre}</p>
      {/* {funcion(2, 3)} */}
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number.isRequired
}

// defaul van antes de los propTypes
// FirstApp.defaultProps = {
//   title: 'No hay t√≠tulo',
//   subtitle: 'No hay subtitulo',
//   nombre: 'Pedro'
// }
