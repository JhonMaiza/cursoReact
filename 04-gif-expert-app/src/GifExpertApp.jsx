import { useState } from "react"
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

  const [ categories, setCategories ] = useState(['One punch']);

  const onAddCategory = ( newCatergory ) => {
    if( categories.includes( newCatergory )) return 
    setCategories( [ newCatergory, ...categories,] );
    // setCategories( cat =>  [...categories, 'valorant'] )
  };
  
  return (
    <>
      <h1>GifExpertApp</h1>

      <AddCategory
        onNewCategory={ onAddCategory }
      />

      { 
        categories.map( category => (
          <GifGrid key={ category } category={category} />
        ))
      }

    </>
  )
}