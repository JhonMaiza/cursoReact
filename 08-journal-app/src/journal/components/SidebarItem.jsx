import { useMemo } from "react";
import { TurnedInNot } from "@mui/icons-material";
import { Grid2, ListItem, ListItemButton, ListItemIcon, ListItemText } from "@mui/material";
import { setActiveNote } from "../../store/journal/journalSlice";
import { useDispatch } from "react-redux";

export const SidebarItem = ( { title = '', body, id, date, imageUrls = [] } ) => {
  const dispatch = useDispatch()

  const newTitle = useMemo( () => {
    return title.length > 17
      ? title.substring(0, 17) + '...'
      : title;
  }, [ title ]);

  const onclickSidebarItem = () => {
    dispatch(setActiveNote( {
      title,
      body,
      id,
      date,
      imageUrls
    }));
  };

  return (
    <ListItem 
      disablePadding
      onClick={ onclickSidebarItem }
    >
        <ListItemButton>
            <ListItemIcon>
                <TurnedInNot />
            </ListItemIcon>
            
            <Grid2 container>
                <ListItemText primary={ newTitle }/>
                <ListItemText secondary={ body }/>
            </Grid2>
        </ListItemButton>
    </ListItem>
  )
}