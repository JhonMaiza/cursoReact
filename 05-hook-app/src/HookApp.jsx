import { CounterApp } from "./01-useState/CounterApp"

export const HookApp = () => {
  return (
    <>
      <h1>
      HooksApp
      </h1>
      <CounterApp/>
    </>
  )
}

