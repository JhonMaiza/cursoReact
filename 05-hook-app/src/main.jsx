import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'
import { HookApp } from './HookApp.jsx'
import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook.jsx'
import { SimpleForm } from './02-useEffect/SimpleForm.jsx'
import { FormWithCustomHook } from './02-useEffect/FormWithCustomForm.jsx'
import { MultipleCustomHook } from './03-examples/MultipleCustomHook.jsx'
import { FocusScreen } from './04-useRef/FocusScreen.jsx'
import { Layout } from './05-useLayoutEffect/Layout.jsx'
import { Memorize } from './06-memos/Memoriza.jsx'
import { MemoHook } from './06-memos/MemoHook.jsx'
import { CallbackHook } from './06-memos/CallbackHook.jsx'
import { Padre } from './07-tarea-memo/Padre.jsx'
import './08-reducer/intro-reducer.js'
import { TodoApp } from './08-reducer/TodoApp.jsx'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <HookApp /> */}
    {/* <CounterWithCustomHook /> */}
    {/* <SimpleForm/> */}
    {/* <FormWithCustomHook/> */}
    {/* <MultipleCustomHook/> */}
    {/* <FocusScreen/> */}
    {/* <Layout/> */}
    {/* <Memorize/> */}
    {/* <MemoHook/> */}
    {/* <CallbackHook/> */}
    {/* <Padre/> */}
    <TodoApp/>
  </React.StrictMode>,
)
