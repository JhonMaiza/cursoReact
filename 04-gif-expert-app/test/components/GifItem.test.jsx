import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components"

describe('Prueba Gifitem', () => {
  const title = 'One piece';
  const url = 'https://ejemplo.com/';

  test('Debe hacer match con el snapshot', () => {
    const { container } = render( <GifItem title={ title } url={ url } />);
    expect( container ).toMatchSnapshot();
  });

  test('debe mostrar la imagen con el url y el alt indicado', () => {
    render( <GifItem title={ title } url={ url } />);
    // screen.debug();
    const { src, alt } = screen.getByRole('img');
    console.log( src, alt);
    
    expect( src ).toBe( url );
    expect( alt ).toBe( title );
  });

  test(' debe mostrar el tÃ­tulo en el componente', () => {
    render( <GifItem title={ title } url={ url } />);
    expect(screen.getByText( title )).toBeTruthy();
  });
})
