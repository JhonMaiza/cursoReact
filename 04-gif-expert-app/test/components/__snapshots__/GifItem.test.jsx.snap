// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba Gifitem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One piece"
      src="https://ejemplo.com/"
    />
    <p>
      One piece
    </p>
  </div>
</div>
`;
